File: tests/fixtures/v2-dynamic-code-flow/positive/DynamicCodeFlow.cs
class DynamicCodeFlow
{
    object Direct(dynamic Request)
    {
        return CSharpScript.EvaluateAsync(Request.Form["code"]);
    }

    object Propagated(dynamic Request)
    {
        var code = Request.Form["code"];
        var forwardedValue = code;
        return CSharpScript.EvaluateAsync(forwardedValue);
    }

    object Restricted(dynamic Request)
    {
        return CSharpScript.EvaluateAsync(
            Request.Form["restricted"],
            ScriptOptions.Default.WithReferences(typeof(SafeGlobals).Assembly));
    }
}

File: tests/fixtures/v2-dynamic-code-flow/positive/dynamic_code_flow.go
package dynamiccodeflow

func direct(r *http.Request) (*plugin.Plugin, error) {
	return plugin.Open(r.FormValue("plugin"))
}

func propagated(r *http.Request) (*plugin.Plugin, error) {
	requested := r.FormValue("plugin")
	alias := requested
	return plugin.Open(alias)
}

func restricted(r *http.Request) (*plugin.Plugin, error) {
	return plugin.Open(filepath.Base(r.FormValue("restricted")))
}
